File: Dev/unitTest_Engine_cpp_gtest/Graphics/PostEffect/PostEffectGaussinaBlur.cpp
#include <Altseed.h>
#include <gtest/gtest.h>
#include <memory>
#include "../../EngineTest.h"


class Graphics_PostEffectGaussianBlur : public EngineTest
{
public:
	Graphics_PostEffectGaussianBlur(bool isOpenGLMode) :
		EngineTest(asd::ToAString("PostEffectGaussianBlur"), isOpenGLMode, 10)
	{}
protected:
	std::shared_ptr<asd::PostEffectGaussianBlur> pe;
	float intensity = 0.0f;
	void OnStart() override
	{
		auto scene = std::make_shared<asd::Scene>();
		auto layer = std::make_shared<asd::Layer2D>();
		auto obj = std::make_shared<asd::TextureObject2D>();
		scene->AddLayer(layer);
		layer->AddObject(obj);
		asd::Engine::ChangeScene(scene);

		auto g = asd::Engine::GetGraphics();
		auto texture = g->CreateTexture2D(asd::ToAString("Data/Texture/Sample1.png").c_str());
		obj->SetTexture(texture);
		obj->SetScale(asd::Vector2DF(1, 1));

		pe = std::make_shared<asd::PostEffectGaussianBlur>();
		pe->SetIntensity(intensity);
		layer->AddPostEffect(pe);

	}
	void OnUpdating() override
	{
		pe->SetIntensity(intensity);
		intensity += 0.5f;
	}
};

ENGINE_TEST(Graphics, PostEffectGaussianBlur)

File: Dev/unitTest_Engine_cs/Graphics/PostEffect/PostEffectGaussinaBlur.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace unitTest_Engine_cs.Graphics.PostEffect
{
	class PostEffectGaussinaBlur : EngineTest
	{
		ace.PostEffectGaussianBlur pe;
		float intensity = 0.0f;

		public PostEffectGaussinaBlur()
			: base(10)
		{
		}

		protected override void OnStart()
		{
			var scene = new ace.Scene();
			var layer = new ace.Layer2D();
			var obj = new ace.TextureObject2D();
			scene.AddLayer(layer);
			layer.AddObject(obj);
			ace.Engine.ChangeScene(scene);

			var g = ace.Engine.Graphics;
			var texture = g.CreateTexture2D("Data/Texture/Sample1.png");
			obj.Texture = texture;
			obj.Scale = new ace.Vector2DF(1, 1);

			pe = new ace.PostEffectGaussianBlur();
			pe.Intensity = intensity;
			layer.AddPostEffect(pe);
		}

		protected override void OnUpdating()
		{
			pe.Intensity = intensity;
			intensity += 0.5f;
		}
	}
}
